pymc3/sampling.py

```diff
@@ -236,6 +238,17 @@ def sample(draws=500, step=None, init='auto', n_init=200000, start=None,
             raise ValueError("Specify only one of step_kwargs and nuts_kwargs")
         step_kwargs = {'nuts': nuts_kwargs}
 
+    if isinstance(step, ExternalEnsembleStepShared):
```

Comment:
What about checking some step method class property? E.g. all steps will have `dependency = 'pymc3'` and emcee will have `dependency = 'emcee'`, that will make possible to check uncommon step method and do necessary imports for trace